### PR TITLE
Give video display as much space as possible

### PR DESCRIPTION
**Pull request by NasalDaemon**
Move room tools to top of page to use full width.
Allow messages to be hidden while video is displaying.
Allow video display to utilise all available width.
Track unread messages while they are hidden.

### CODE REVIEW

**Comment by jeremyckahn** [comment]
Great work so far @NasalDaemon! This is already working very well. I noticed that this PR is still in "draft" state so I'll hold off on reviewing it until it's marked as ready for review. Let me know if you'd like support with anything in the meantime.

**Comment by NasalDaemon** [comment]
> Great work so far @NasalDaemon! This is already working very well. I noticed that this PR is still in "draft" state so I'll hold off on reviewing it until it's marked as ready for review. Let me know if you'd like support with anything in the meantime.

Thanks @jeremyckahn! I feel I've polished it enough that it's worth looking at, so I've taken it out of draft state.

File: src/components/Shell/ShellAppBar.tsx
Comment (jeremyckahn) [inline src/components/Shell/ShellAppBar.tsx:8]: Why did we switch from default to named imports here? [At least for development, it seems that using default imports is faster](https://mui.com/material-ui/guides/minimizing-bundle-size/#development-environment).

**Review by jeremyckahn — CHANGES_REQUESTED** [review]
Excellent work @NasalDaemon! Thanks for taking the time to make this change. It's a significant improvement to the UI for video calls! I think it'll help people take Chitchatter more seriously for that use case. 🙂

There were a few things I wanted to revisit before merging, but nothing major. Let me know if you have any questions!

File: src/components/Shell/Shell.tsx
Comment (jeremyckahn) [inline src/components/Shell/Shell.tsx:159]: It looks like this is causing the TypeScript build to break:

```
src/components/Shell/Shell.tsx(153,27): error TS2339: Property 'mozCancelFullScreen' does not exist on type 'Document'.
src/components/Shell/Shell.tsx(154,18): error TS2339: Property 'mozCancelFullScreen' does not exist on type 'Document'.
src/components/Shell/Shell.tsx(155,27): error TS2339: Property 'webkitExitFullscreen' does not exist on type 'Document'.
src/components/Shell/Shell.tsx(156,18): error TS2339: Property 'webkitExitFullscreen' does not exist on type 'Document'.
src/components/Shell/Shell.tsx(157,27): error TS2551: Property 'msExitFullScreen' does not exist on type 'Document'. Did you mean 'exitFullscreen'?
src/components/Shell/Shell.tsx(158,18): error TS2551: Property 'msExitFullScreen' does not exist on type 'Document'. Did you mean 'exitFullscreen'?
```

The logic here seems sound, we just need keep the TypeScript compiler satisfied.

File: src/components/Room/RoomShowMessagesControls.tsx
Comment (NasalDaemon) [reply src/components/Room/RoomShowMessagesControls.tsx]: been playing around with the colours and trying to keep them consistent in their meaning. I've settled on showing red to hide the messages, and green when they are showing, just like for audio/video

File: src/components/Room/Room.tsx
Comment (NasalDaemon) [reply src/components/Room/Room.tsx:73]: I'm not really sure what the cheapest implementation is to listen to window size changes, this is just something I found in a google search. any better suggestions?

File: src/components/Room/RoomShowMessagesControls.tsx
Comment (NasalDaemon) [reply src/components/Room/RoomShowMessagesControls.tsx]: It's not what you suggested, but I think being consistent is quite important in reducing users' mental overhead.

File: src/components/Room/Room.tsx
Comment (jeremyckahn) [reply src/components/Room/Room.tsx:73]: There are a variety of minimal ways to achieve this. It would be ideal to use an external implementation that we could pull in as an NPM dependency. I found this package that seems like it might be worth giving a try: https://www.npmjs.com/package/@react-hook/window-size. According to [Bundlephobia](https://bundlephobia.com/package/@react-hook/window-size@3.1.1) it's only 1.6kB, so it would not impact our bundle size significantly.

If that doesn't work for whatever reason, this implementation seems like a pretty copy/paste-able solution: https://usehooks.com/useWindowSize/

File: src/components/Room/RoomShowMessagesControls.tsx
Comment (jeremyckahn) [inline src/components/Room/RoomShowMessagesControls.tsx:32]: This is an excellent touch that I think users will really appreciate! :)

File: src/components/Shell/ShellAppBar.tsx
Comment (jeremyckahn) [inline src/components/Shell/ShellAppBar.tsx:189]: I didn't know about this MUI component. Very handy!

File: src/components/Shell/Shell.tsx
Comment (jeremyckahn) [inline src/components/Shell/Shell.tsx:168]: While testing this PR, I noticed this error being thrown a few times:

```
Uncaught (in promise) TypeError: Not in fullscreen mode
    exitFullscreen Shell.tsx:161
    Shell Shell.tsx:177
```

It suggests that this function is getting called more often than it should. It might be worth adding some conditional logic around the call site to avoid invoking the function unnecessarily, or potentially wrapping the body of this function in a `try`/`catch` to silence the error if it is unavoidable.

**Review by jeremyckahn — CHANGES_REQUESTED** [review]
This is some really excellent work @NasalDaemon. Thanks for continuing to iterate on this! The UX feels great and everything is working smoothly. There were just a few minor things I wanted to discuss about the code. The item I'm most concerned about is seeing if we can swap out the React Native dependencies for a more lightweight alternative solution. Once that's sorted out I think this should be ready to merge.

**Review by jeremyckahn — APPROVED** [review]
Thanks for making all the changes and sticking with this @NasalDaemon! This is a significant improvement to Chitchatter's video call experience. I'm sure that users are going to love this! 👏

I'll merge and release this now.

**Pull request merged**
